vtep: Documents VTEP redirection and L7 policy partial incompatibility (#19700)
cilium-team/cilium: Documentation/gettingstarted/vtep.rst

.. warning::

>    This feature is in beta, and is currently partially incompatible with L7 policy.

Reviewer: ```suggestion
   This feature is in beta, and currently, it is partially incompatible with the L7 policy.
```

>    When pod with egress L7 policy sending request to VTEP devices, the VTEP redirection

Reviewer: ```suggestion
   When pod with egress L7 policy sends a request to VTEP devices, the VTEP redirection
```

>    When pod with egress L7 policy sends a request to VTEP devices, the VTEP redirection

Reviewer 2: ```suggestion
   When a pod with an egress L7 policy sends a request to VTEP devices, the VTEP redirection
```
